### PR TITLE
Move initcontainer job out of Solidity workflow to a separate workflow

### PR DESCRIPTION
**Pull request by michalinacienciala**
In a separate PR (#873) we're changing the kube configuration for
`keep-ecdsa` pods, so that the population of `TBTCSystem` address would
happen automatically. For the feature to work correctly we need to build
the `keep-ecdsa` initcontainer after migrating `tbtc` contracts. To do
that, we need to have separate workflows for `keep-ecdsa` contracts
migration and for initcontainer building/publishing - we're doing that
separation in this PR.
Apart from that, we'll be changing the CI flow configuration in the
`keep-network/ci/config/config.json` which orchestrates the order of
workflows execution during deployment on testnet.

Refs:
#873
https://github.com/keep-network/ci/pull/12

TODO:
- [x] remove the `--wip` suffix from the image name once tested

### CODE REVIEW

**Comment by michalinacienciala** [comment]
Workflow verifed on slightly altered configuration (some of the variables have been hardcoded), see https://github.com/keep-network/keep-ecdsa/runs/3201938191?check_suite_focus=true. Published image: [link](https://console.cloud.google.com/gcr/images/keep-test-f3e0/GLOBAL/initcontainer-provision-keep-ecdsa-ethereum-wip@sha256:5f07f4848382044e391856c9b7a4f9a901300accdcba3ca344766a9db443cdae/details?tab=info&authuser=4&folder&organizationId&supportedpurview=project).

File: .github/workflows/initcontainer.yml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+      - uses: actions/checkout@v2
+
+      - name: Load environment variables
+        uses: keep-network/load-env-variables@v1
+        with:
+          environment: ${{ github.event.inputs.environment }}
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "12.x"
```
Comment (nkuba) [inline .github/workflows/initcontainer.yml:34]: Can we switch to `14` here?

File: .github/workflows/initcontainer.yml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+      - name: Cache node modules
```
Comment (nkuba) [inline .github/workflows/initcontainer.yml:36]: We're not using `actions/setup-node` cache due to a non-root path here, right? Can we add a comment, perfectly with a link to an issue describing the problem to track when we will be able to switch caching?

File: .github/workflows/initcontainer.yml [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+          upstream-builds: ${{ github.event.inputs.upstream_builds }}
+          query: |
+            keep-core-contracts-version = github.com/keep-network/keep-core/solidity#version
+            keep-ecdsa-contracts-version = github.com/keep-network/keep-ecdsa/solidity#version
+
+      - name: Resolve latest contracts
+        run: |
+          npm install --save-exact \
+            @keep-network/keep-core@${{ steps.upstream-builds-query.outputs.keep-core-contracts-version }} \
+            @keep-network/keep-ecdsa@${{ steps.upstream-builds-query.outputs.keep-ecdsa-contracts-version }}
```
Comment (nkuba) [inline .github/workflows/initcontainer.yml:61]: Don't we need to pull tbtc contracts too to obtain `TBTCSystem` artfact for https://github.com/keep-network/keep-ecdsa/pull/873? Or are we planning to do that in a separate PR?

File: .github/workflows/initcontainer.yml [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.GCR_REGISTRY_URL }}
+          username: _json_key
+          password: ${{ secrets.KEEP_TEST_GCR_JSON_KEY }}
+
+      - name: Build and publish initcontainer
+        uses: docker/build-push-action@v2
+        env:
+          IMAGE_NAME: initcontainer-provision-keep-ecdsa-${{ env.CHAIN }}-wip #TODO: remove `--wip` before merging to main
```
Comment (nkuba) [inline .github/workflows/initcontainer.yml:91]: Leaving a comment here to remember to remove `-wip`.

File: .github/workflows/initcontainer.yml (hunk 3 above)
Comment (michalinacienciala) [reply .github/workflows/initcontainer.yml]: You're right, we do. Added in dffdbc1cce9b49299f77dc85eafd9e4feeafdf3d.

File: .github/workflows/initcontainer.yml (hunk 4 above)
Comment (michalinacienciala) [reply .github/workflows/initcontainer.yml]: I've removed it now.

File: .github/workflows/initcontainer.yml [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+            ${{ runner.os }}-
+
+      - name: Get upstream packages' versions
+        uses: keep-network/ci/actions/upstream-builds-query@v1
+        id: upstream-builds-query
+        with:
+          upstream-builds: ${{ github.event.inputs.upstream_builds }}
+          query: |
+            keep-core-contracts-version = github.com/keep-network/keep-core/solidity#version
+            keep-ecdsa-contracts-version = github.com/keep-network/keep-ecdsa/solidity#version
```
Comment (nkuba) [inline .github/workflows/initcontainer.yml:60]: Aren't we missing `tbtc-contracts-version` assignment here?

File: .github/workflows/initcontainer.yml [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+          push: true
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache-new
+
+      - name: Notify CI about completion of the workflow
+        uses: keep-network/ci/actions/notify-workflow-completed@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.CI_GITHUB_TOKEN }}
+        with:
+          module: "github.com/keep-network/keep-ecdsa-initcontainer"
```
Comment (nkuba) [inline .github/workflows/initcontainer.yml:116]:
```suggestion
          module: "github.com/keep-network/keep-ecdsa/initcontainer"
```

File: .github/workflows/initcontainer.yml [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+          url: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}
+          environment: ${{ github.event.inputs.environment }}
+          upstream_builds: ${{ github.event.inputs.upstream_builds }}
+          upstream_ref: ${{ github.event.inputs.upstream_ref }}
+          version: ${{ needs.contracts-migrate-and-publish.outputs.version }}
```
Comment (nkuba) [inline .github/workflows/initcontainer.yml:121]: `contracts-migrate-and-publish.outputs.version` was removed from the contracts workflow. Maybe we could use `steps.upstream-builds-query.outputs.keep-ecdsa-contracts-version` instead?

**Review by nkuba — CHANGES_REQUESTED** [review]

File: .github/workflows/initcontainer.yml (hunk 5 above)
Comment (michalinacienciala) [reply .github/workflows/initcontainer.yml:60]: You're right, added in a5bd2779a91e567ead8cc5685407156c9ddbe922.

File: .github/workflows/initcontainer.yml (hunk 7 above)
Comment (michalinacienciala) [reply .github/workflows/initcontainer.yml]: Good catch and the suggestion. Fixed in 2848b682da61d45d64a1bd9ba045d3cd7b3ddf80.

**Review by nkuba — APPROVED** [review]
